tests/integration_tests/test_detect_db_type_and_version.py

```
@@ -12,6 +11,36 @@
+        assert version1 == version3, f"Expected {version1} == {version3}"
+        
+        # Test 3-part version (should be normalized to 4 parts)
+        version4 = Version("1.2.3")
+        version5 = Version("1.2.3.0")
+        assert version4 == version5, f"Expected {version4} == {version5}"
+        
+        # Test string representation (preserve all parts including trailing .0)
+        assert str(version1) == "1.0.1.0"
+        assert str(version4) == "1.2.3.0"  # Full version preserved
+        
+        print(f"\n✅ Version comparison tests passed")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)